src/components/delivery/Compass.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { selectRoverPosition } from "../../store/telemetrySlice";
+import "./Compass.css";
+
+const Compass = () => {
+  const { orientW, orientX, orientY, orientZ, posX, posY, posZ } = useSelector(
+    selectRoverPosition
+  );
+
+  let needleColor;
+  let pitch;
+  let roll;
+  
+  if(pitch == null && roll == null) {
```

Comment:
Did you test this? This defines `pitch` and `roll` to be undefined and then immediately nullchecks. So this won't work when you want the compass to actually display things